### PR TITLE
Process DecisionRequirements DELETED events

### LINKED ISSUE #9773: Handle a DecisionRequirements Deleted event
# Description
We need to create an event applier to handle the Process Deleted event.

- Create a `DecisionRequirementsDeletedApplier`
- If deleted version is the latest version:
    - Find new latest version:
        - If found: update `latestDecisionRequirementsKeysById`
        - If not found: delete from `latestDecisionRequirementsKeysById`
- Delete the decision requirements from the state
    - Delete from `decisionRequirementsByKey`
    - Delete from `decisionRequirementsKeyByIdAndVersion`

Blocked by:
- #9767

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR adds an EventApplier for `DecisionRequirements ` events. The applier will delete the decision from the state.

Since we store the latest `decisionRequirementsKey` for a `decisionRequirementsId` in the `latestDecisionRequirementsKeysById` ColumnFamily we have a two scenarios here:

1. The deleted DRG is the lates. At this point we must figure out what to do. If there are DRGs with lower versions we must update this ColumnFamily to refer to the previous version. If there are no other versions we must delete the DRG from this ColumnFamily.
2. The deleted DRG is **not** the latest decision. We can completely ignore the `latestDecisionRequirementsKeysById`. This columnFamily refers to a DRG that is not related to the DRG we are deleting.

The DRG will be deleted from all other relevant ColumnFamilies.

## Related issues

closes #9773 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableDecisionState.java [hunk 1]
```diff
@@ -35,4 +35,6 @@ public interface MutableDecisionState extends DecisionState {
    * @param record the record of the decision
    */
   void deleteDecision(DecisionRecord record);
+
```
Comment (koevskinikola) [inline engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableDecisionState.java:38]: :wrench: Should we also add some Javadocs here, like in the methods above?

Maybe something like the following:
```suggestion
/**
   * Deletes a decision requirements (DRG) from the state. Updates the latest version of the decision requirements (DRG) if the deleted
   * version is the latest version and a previous version is available.
   *
   * @param record the record of the decision requirements (DRG)
   */
```

**Review by koevskinikola — CHANGES_REQUESTED** [review]
@remcowesterhoud the changes look good. Just a small hint regarding Javadocs.

File: engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableDecisionState.java (hunk 1 above)
Comment (remcowesterhoud) [reply engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableDecisionState.java:38]: Oops, definitely. Nice catch 👍

**Comment by remcowesterhoud** [comment]
Btw @koevskinikola. When you do a review and it's a small thing like some missing javadoc feel free to just approve it already and tell me to fix it. There's most likely no new review required, but now you'll have to spend time looking at this again after my fix 😄

**Review by koevskinikola — APPROVED** [review]
All good! :rocket: 

bors merge

**Pull request merged**